service/status-report/main.go

```diff
@@ -138,10 +141,46 @@ func main() {
 	status := providerclient.NewStorageClientStatus().
 		SetPlatformVersion(pltVersion).
 		SetOperatorVersion(oprVersion)
-	if _, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status); err != nil {
+	statusResponse, err := providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status)
+	if err != nil {
 		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 
+	versionFromResponse, err := getVersionFromChannel(statusResponse.DesiredClientOperatorChannel)
+	if err != nil {
+		klog.Exitf("Failed to map channel name from status response to version: %v", err)
+	}
+
+	var updateAnnotation bool
+	if storageClient.GetAnnotations()[desiredSubscriptionChannelAnnotationKey] == "" {
+		updateAnnotation = true
+	}
+
+	var versionFromAnnotation *version.Version
+	if !updateAnnotation {
+		var err error
+		versionFromAnnotation, err = getVersionFromChannel(storageClient.GetAnnotations()[desiredSubscriptionChannelAnnotationKey])
+		if err != nil {
+			klog.Exitf("Failed to map channel name from storageclient annotation to version: %v", err)
+		}
+		// bump the annotation if it is less than version value from response, having higher version in annotation implies user overrode the upgrade
+		// negation is used here for checking strictly greater version
+		if !versionFromResponse.LessThan(versionFromAnnotation) {
+			// since we are referring only version from channel, by checkig for strictly greater version we'll enforce the channel name is also same
+			updateAnnotation = true
+		}
```

Comment:
This entire check does not belong here. You should always store the desired channel. The code that uses the annotation should take into consideration the value itself.

Reply:
The intention was for user to provide a way to bump the version by overriding the annotation, if we are updating the annotation w/ a lower version (as wanted by provider) then users intention will not stick, isn't it?

Reply:
Ok, but why is the version in the channel an indication that it was user-edited? Also, how are you expecting the user to understand the channel text format?

Reply:
> but why is the version in the channel an indication that it was user-edited
- we weren't going to update the annotation value if it is more than provider version and that signifies it's user edited

> how are you expecting the user to understand the channel text format
- the channel name is usually version prefixed w/ `stable-`, since there isn't any documentation that it'll stay the same dropping this as suggested offline.

Reply:
Removed code which extracts version from channel.